processing/src/main/java/org/apache/druid/segment/TimeAndDimsPointer.java

```diff
@@ -94,7 +94,7 @@ public class TimeAndDimsPointer implements Comparable<TimeAndDimsPointer>
     this.timestampSelector = timestampSelector;
     this.timePosition = timePosition;
     Preconditions.checkArgument(
-        timePosition >= 0 && timePosition <= dimensionSelectors.length,
+        timePosition <= dimensionSelectors.length,
```

Comment:
I suppose `timePosition` can be `-1` for projections, so part of this check had to go. Please update the message too.